components/builder-depot/src/server.rs

```diff
@@ -457,6 +457,29 @@ fn write_archive(filename: &PathBuf, body: &mut Body) -> Result<PackageArchive>
     Ok(PackageArchive::new(filename))
 }
 
+fn generate_origin_encryption_keys(req: &mut Request) -> IronResult<Response> {
+    debug!("Generate Origin Encryption Keys {:?}", req);
+    let session = req.extensions.get::<Authenticated>().unwrap().clone();
```

Comment:
If you're concerned about minimizing your allocations :trollface: you could eliminate this line completely and pull the session out of the `req` struct in your `generate_origin_encryption_keys` helper function.  Doing it there would work without cloning it.